### PR TITLE
[pt2] add meta and `SymInt` support for `linalg_lu`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101375
* __->__ #101372
* #101367
* #101362
* #101315

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Part of this stack because it unlocks tests next to the prev PR in the stack.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -604,6 +604,34 @@ def linalg_ldl_solve_meta(
...
+
+    sizes[-1] = k
+    L = A.new_empty(sizes)
+
+    sizes[-2] = k
+    sizes[-1] = n
+    U = A.new_empty(sizes)
+    return P, L, U
+
+
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:634]: See `TORCH_META_FUNC(linalg_lu)` in core.

**Review by lezcano — APPROVED** [review]

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**